control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -810,7 +810,8 @@ func (r *HostedControlPlaneReconciler) healthCheckKASLoadBalancers(ctx context.C
 		if hcp.Spec.Platform.Type == hyperv1.IBMCloudPlatform {
 			port = config.KASSVCIBMCloudPort
 		}
-		if hcp.Spec.Platform.Type == hyperv1.AzurePlatform {
+		if hcp.Spec.Platform.Type == hyperv1.AzurePlatform ||
+			hcp.Spec.Platform.Type == hyperv1.KubevirtPlatform && hcp.Annotations[hyperv1.ManagementPlatformAnnotation] == string(hyperv1.AzurePlatform) {
```

Comment:
I don't think the `hcp.Spec.Platform.Type == hyperv1.KubevirtPlatform` is really necessary here. If the platform is azure, or the mgmt cluster is being hosted on azure, then this is the correct path regardless if kubevirt is involved.

Reply:
fixed.